### ISSUE TITLE
Not passing in the attention mask breaks attention

### ISSUE THREAD

**Issue by sgligorijevicTT** [tenstorrent/tt-xla#434]
When the attention mask is not passed into transformers models, a code path with `jnp.ones_like` is hit. Lowering of `ones_like` involves an add operation with broadcasting, which hits a metal bug. As a result, all zeros mask is generated and attention layer returns wrong results. This ruins the accuracy of all transformer models.
A workaround is to pass in the attention mask.

Should *hopefully* be fixed by https://github.com/tenstorrent/tt-metal/pull/17160 . Update: the relevant PR was merged and then reverted, I am tracking what's happening in metal and will link the new PR as soon as it is opened.
Metal already has multiple issues opened regarding broken broadcasting in binary ops and is actively working on binary_ng.
